### DISCUSSION TITLE
Using Data Views in your plugin/preparing for the future

### DISCUSSION THREAD

**Discussion by annezazu** [WordPress/developer-blog-content#232] [Topic Ideas]
[With Data Views](https://github.com/WordPress/gutenberg/issues/55083) making its first entry into the WordPress sphere in 6.5 in a fairly contained way, it feels like a good time to start talking about what's to come and ensuring plugin authors can start reusing what's being done in Core to level up their own designs/flows. I think this would make for a good developer blog post perhaps going through what's available in core, how to follow along, and what components are available to use.

**Comment by justintadlock** [comment]
There are some concerns around approving this topic right now. There's a general agreement is that the concept is good, but it could use a little fleshing out from a writer who's willing to work on it. A few of our normal members were out of the meeting and may go ahead and vote to approve in the next week.

My feedback from the March Editorial Group meeting:

> For the Data Views discussion, I like the idea in general. But I’m hesitant to approve it because there’s no specific writer with a more fleshed out idea for what the post might cover.

I'm also concerned that when there's no writer already attached to a discussion topic that no one ends up picking it up after approval. Outside of that, I want us to get this approved.

Other feedback from the meeting included:

> It could also be looked at as a follow-up article for https://github.com/WordPress/developer-blog-content/issues/191, where more recent/experimental components are explored - for example, the Data Views.

**Comment by annezazu** [comment]
Sharing this as inspiration/where we might be able to pull a writer from https://twitter.com/dashkitHQ/status/1762849625210249531?s=20

**Comment by annezazu** [comment]
Noting this for the future when a post like this will be helpful: https://github.com/WordPress/gutenberg/issues/61084

**Comment by juanmaguitar** [comment]
I would be up to write such a post, starting in Q3.

**Comment by bph** [comment]
@juanmaguitar Awesome that you join u on the work for the Developer Blog! I am excited to read your insights on how developer can take advantage of the Dataviews.  

Writers can help the approval process along by providing a ['Fat outline'](https://bernoff.com/blog/fat-outline) (Link to Jeff Bernoff's blog post) so convey the nature or the approach for a particular post. 

Editorial Meetings are held every first Thursday of the month at 13:00 UTC
Here is the list of Q3 meetings

- June 6th, 2024
- July 4th, 2024
- August 1, 2024

**Reply by juanmaguitar** [reply]
Thanks for this info @bph 
I still need to check what's going to be available for WP 6.6 regarding DataViews (I'd like to have a conversation with @oandregal about this in the following weeks)

Very roughly,  I was thinking about explaining the creation of a plugin that displays a table with the [`@wordpress/dataviews`](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-dataviews/). The content of that table can vary in complexity from a static JSON (like [this one](https://github.com/WordPress/block-development-examples/blob/trunk/_data/examples.json)) to dynamic data (like the type of data explained in [this tutorial](https://developer.wordpress.org/block-editor/how-to-guides/data-basics/)). 

The purpose of this article would be to show:
- What is "Data Views" and examples of where is being used in the Site Editor
- how to use the `@wordpress/dataviews` to display data in a plugin (import, load and setup)
- how to load different types of data
- how to get different views for this data
- how to filter and sort this data

I still need to do some research, but I'd like to share a better ['Fat outline'](https://bernoff.com/blog/fat-outline) (love this concept, by the way 🙂) ready by July 4th, 2024, or August 1, 2024

**Reply by oandregal** [reply]
> ensuring plugin authors can start reusing what's being done in Core to level up their own designs/flows

I've given a talk recently about this for a developer-savvy audience. It was demo-heavy and I presented two demos: use `@wordpress/block-editor` to build your own custom editor & use `@wordpress/dataviews` to build your own data-driven pages.

The relevant parts for DataViews:

- Talk https://oandre.gal/talks/wordpress-at-index-gal/#7 (you can use arrow keys to navigate)
- The repo with the demos https://github.com/oandregal/index-gal-wordpress/tree/trunk/dataviews Switch to the `*dataviews*` branches to see examples.
- YouTube https://youtu.be/3FDg5kyBPEU?si=eySMpUn1vkrYateN&t=2430 starting at the point where I present DataViews and do the demo. It's in Galician but a Spanish speaker should be able to follow it.

As the audience was outside the WordPress ecosystem, the demos were focused on how you can use these even if you don't use WordPress at all.

I'm happy to help and provide feedback, though I'll have more availability by end of May/first of June.

**Comment by bph** [comment]
This topic has been approved. Please continue the conversation on the issue

- #273

**Comment by juanmaguitar** [comment]
Post published!
https://developer.wordpress.org/news/2024/08/27/using-data-views-to-display-and-interact-with-data-in-plugins/

**Comment by juanmaguitar** [comment]
A second part has been published at: https://developer.wordpress.org/news/2024/09/23/actions-from-data-views-adding-images-to-the-media-library/
